package/kedro_viz/server.py

```diff
@@ -317,90 +330,143 @@ def _construct_layer_mapping(catalog):
     return dataset_to_layer
 
 
-# pylint: disable=too-many-locals
-def format_pipeline_data(pipeline, catalog):
+def _pretty_name(name: str) -> str:
+    name = name.replace("-", " ").replace("_", " ")
+    parts = [n[0].upper() + n[1:] for n in name.split()]
+    return " ".join(parts)
+
+
+def format_pipelines_data(pipelines: Dict[str, "Pipeline"], catalog) -> Dict[str, list]:
     """
-    Format pipeline and catalog data from Kedro for kedro-viz
+    Format pipelines and catalog data from Kedro for kedro-viz.
 
     Args:
-        pipeline: Kedro pipeline object
-        catalog:  Kedro catalog object
-    """
+        pipelines: Dictionary of Kedro pipeline objects.
+        catalog: Kedro catalog object.
 
-    def pretty_name(name):
-        name = name.replace("-", " ").replace("_", " ")
-        parts = [n[0].upper() + n[1:] for n in name.split()]
-        return " ".join(parts)
+    Returns:
+        Dictionary of pipelines, nodes, edges, tags and layers, and pipelines list.
 
+    """
+    pipeline_list = []
+    # keep track of a sorted list of nodes to returned to the client
+    node_list = []
+    # keep track of edges in the graph: [{source_node_id -> target_node_id}]
+    edge_list = []
     # keep tracking of node_id -> node data in the graph
     nodes = {}
-    # keep track of a sorted list of nodes to returned to the client
-    sorted_nodes = []
     # keep track of node_id -> set(child_node_ids) for layers sorting
     node_dependencies = defaultdict(set)
-    # keep track of edges in the graph: [{source_node_id -> target_node_id}]
-    edges = []
+    tags = set()
+
+    for pipeline_key, pipeline in pipelines.items():
+        pipeline_list.append({"id": pipeline_key, "name": _pretty_name(pipeline_key)})
```

Comment:
Do we need `id` and prettified `name` for pipelines list?

Reply:
That's what @richardwestenra requested :)